File: Python/2016/Hidden Palindrome.py
# J3

word = (input())
length1 = 0
sub_words = []


def reverse(x):
    return x[::-1]


def check(word1):
    if word1 == reverse(word1):
        return True
    else:
        return False


for i in range(len(word) + 1):
    for f in range(len(word) + 1):
        modiword = word[i:f]
        sub_words.append(modiword)

for i in sub_words:
    z = check(i)
    y = len(i)
    if z and y >= length1:
        length1 = len(i)


print(length1)

File: Java/2016/HiddenPalindrome.java
// J3

import java.util.ArrayList;
import java.util.Scanner;

public class HiddenPalindrome {

    public static String reverse(String str) {
        StringBuilder builder = new StringBuilder();
        builder.append(str);
        return builder.reverse().toString();
    }

    public static boolean check(String word) {
        return (word.equals(reverse(word)));
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int length = 0;

        String word = scanner.next();

        ArrayList<String> subwords = new ArrayList<String>();

        for (int i = 0; i < word.length() + 1; i++) {
            for (int f = 0; f < word.length() + 1; f++) {
                try {
                String modiword = word.substring(i, f);
                subwords.add(modiword); 
                }
                catch (Exception e) {
                    ;
                }
            }
        }

        for (int i = 0; i < subwords.size(); i++) {
            String subword = subwords.get(i);
            if (check(subword) && subword.length() > length) length = subword.length();
        }

        System.out.println(length);

        scanner.close();
    }    
}